seo.coffee

```diff
@@ -154,23 +173,24 @@ SEO =
 escapeHtmlAttribute = (string) ->
   return ("" + string).replace(/'/g, "&apos;").replace(/"/g, "&quot;")
 
-getCurrentRouteName = ->
-  router = Router.current()
-  return unless router
-  routeName = router.route.getName()
-  return routeName
-
-# Get seo settings depending on route
-Deps.autorun( ->
-  currentRouteName = getCurrentRouteName()
-  return unless currentRouteName
-  Meteor.subscribe('seoByRouteName', currentRouteName)
-)
-
-# Set seo settings depending on route
-Deps.autorun( ->
-  return unless SEO
-  currentRouteName = getCurrentRouteName()
-  settings = SeoCollection.findOne({route_name: currentRouteName}) or {}
-  SEO.set(settings)
-)
+Meteor.startup () ->
+  getCurrentRouteName = ->
+    router = Router.current()
```

Comment:
No, because Router is aliased to FlowRouter in order to mung the API. I have
been running this code in production on my website for 9 months with 0 errors,
so I'm pretty sure it works :)

Greg


Sent using CloudMagic Email
[https://cloudmagic.com/k/d/mailapp?ct=pa&cv=8.0.91&pv=5.1.1&source=email_footer_2] On Wed, Jan 27, 2016 at 1:23 PM, Daniel Bernhard < notifications@github.com [notifications@github.com] > wrote:
In seo.coffee [https://github.com/DerMambo/ms-seo/pull/67#discussion_r51035400] :

> -Deps.autorun( ->
> -  currentRouteName = getCurrentRouteName()
> -  return unless currentRouteName
> -  Meteor.subscribe('seoByRouteName', currentRouteName)
> -)
> -
> -# Set seo settings depending on route
> -Deps.autorun( ->
> -  return unless SEO
> -  currentRouteName = getCurrentRouteName()
> -  settings = SeoCollection.findOne({route_name: currentRouteName}) or {}
> -  SEO.set(settings)
> -)
> +Meteor.startup () ->
> +  getCurrentRouteName = ->
> +    router = Router.current()


Shouldn't this be FlowRouter.current().route.name ?

—
Reply to this email directly or view it on GitHub [https://github.com/DerMambo/ms-seo/pull/67/files#r51035400] .[https://github.com/notifications/beacon/AAF_yvImIMqHxh4uR4XOCdTRLmRGSWvlks5peRCtgaJpZM4FrbpP.gif]